File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Ruby/drag_and_drop_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class DragAndDropTest < Test::Unit::TestCase
	def setup
		
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/DragDropDemo.html'
		@verification_errors = []
	end
	
	def test_drag_and_drop
		source = @driver.find_element :id => "draggable"
		target = @driver.find_element :id => "droppable"
				
		@driver.action.drag_and_drop(source, target).perform
		assert_equal "Dropped!",target.text
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end

File: XX - Backup Project/xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Python/drag_and_drop_test.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from array import *
import time, unittest

class DragAndDropTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		
	def test_drag_and_drop(self):
		driver = self.driver
		driver.get("http://dl.dropbox.com/u/55228056/DragDropDemo.html")
		
		source = driver.find_element_by_id("draggable")
		target = driver.find_element_by_id("droppable")
				
		ActionChains(self.driver).drag_and_drop(source, target).perform()
		self.assertEqual("Dropped!", target.text)
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()